### ISSUE TITLE
[data grid] Please add edit example for treedata

### ISSUE THREAD

**Issue by entaildevops** [mui/mui-x#15420]
### Related page

https://mui.com/x/react-data-grid/tree-data/

### Kind of issue

Missing information

### Issue description

It is currently quite unclear how row editing should be accomplished properly with treedata. If you the [full featured crud example](https://mui.com/x/react-data-grid/editing/#full-featured-crud) is modified to use treedata, the editing works - BUT: every edit causes the row to collapse, which is very bad UX. The collapsing occurs because the example uses processRowUpdate to handle updates. As stated [here ](https://mui.com/x/react-data-grid/row-updates/#the-updaterows-method) apiref.current.updateRows() would not cause this. But with a lack of example, it is a bit hard to get grasp on how to do it. 

The Full example of treedata should have example also on how to properly edit the rows with decent UX when treedata is present.

**Search keywords**: treedata

**Comment by michelengelen** [comment]
Hey @entaildevops ... That's a good request.
I'll add this to our umbrella issue to add recipes for specific use cases.
Thanks for raising this.

The normal editing feature works though, so what exactly are you trying to accomplish? If you can explain your use case we might be able to help you with it.

**Comment by entaildevops** [comment]
Good to hear that you´re going to create more examples. 

Like wrote, the problem is that when editing any expanded row (child of another row), the expanded state is lost and collapses. I _think_ this is because the row updating occurs inside processRowUpdate (like it is also done in the [CRUD example](https://mui.com/x/react-data-grid/editing/#full-featured-crud)). So an example where the collapsing does not occur would be helpful!

**Comment by michelengelen** [comment]
> Good to hear that you´re going to create more examples, @michelengelen!
> 
> However, I was able to solve the problem with collapsing rows: the problem was me using setRows() to update all rows within processRowUpdate, which caused the rows collapsing (just as the docs imply). The solution was to use apiRef.current.updateRows() to update only affected rows instead.
> 
> Still, it would be helpful if the treedata example would include editing capabilities.

Glad to hear that you could fix it on your own. I would have been able to point you to that as well if you had provided a reproduction example :P

I'll keep that in mind for the next time.

@mui/xgrid should we maybe encourage the use of `updateRows` instead of rows in state in the recipes more? Feels like this comes up every now and then!

**Comment by entaildevops** [comment]
> @mui/xgrid should we maybe encourage the use of updateRows instead of rows in state in the recipes more? Feels like this comes up every now and then!

I agree. People tend to copy/paste examples and go from there, so the "correct" way should be used as much as possible. At least I had been using DataGrid for a good while before I bumped into the fact that setRows() is NOT the recommended way to update your content. Yes, the correct way is is mentioned in the docs, but it is easy to miss I guess =)

**Comment by siriwatknp** [comment]
Since this issue is 1 year old and seems like the expansion issue has been fixed in the past. I think we can close this one.

Based on this [sandbox](https://stackblitz.com/edit/q7cjysy9?file=src%2FDemo.tsx) I forked from the [full CRUD](https://mui.com/x/react-data-grid/editing/#full-featured-crud) of the docs, adapting to tree data is straight forward. I don't think another recipe need to be added.

What do you think? cc @michelengelen 

This, however, is a different topic
> should we maybe encourage the use of updateRows instead of rows in state in the recipes more? Feels like this comes up every now and then!

**Comment by arminmeh** [comment]
[This](https://github.com/mui/mui-x/pull/19697) made the grid keep the expansion state on updates.
Released in [8.13.0](https://github.com/mui/mui-x/releases/tag/v8.13.0)
